src/java/org/apache/cassandra/io/sstable/format/SSTableReader.java

```diff
@@ -1125,18 +839,47 @@ private SSTableReader cloneAndReplace(DecoratedKey newFirst, OpenReason reason,
                                                  metadata,
                                                  ifile != null ? ifile.sharedCopy() : null,
                                                  dfile.sharedCopy(),
-                                                 newSummary,
                                                  bf.sharedCopy(),
                                                  maxDataAge,
                                                  sstableMetadata,
                                                  reason,
-                                                 header);
+                                                 header,
+                                                 newSummary);
```

Comment:
nit: I suppose we could minimize the diff here by leaving the argument order (for `newSummary`) the same.

Reply:
+1